yarn.lock

```diff
@@ -6194,27 +6194,32 @@ __metadata:
   languageName: node
   linkType: hard
 
-"cheerio@npm:^0.22.0":
-  version: 0.22.0
-  resolution: "cheerio@npm:0.22.0"
+"cheerio-select@npm:^2.1.0":
+  version: 2.1.0
+  resolution: "cheerio-select@npm:2.1.0"
+  dependencies:
+    boolbase: ^1.0.0
+    css-select: ^5.1.0
+    css-what: ^6.1.0
+    domelementtype: ^2.3.0
+    domhandler: ^5.0.3
+    domutils: ^3.0.1
+  checksum: 843d6d479922f28a6c5342c935aff1347491156814de63c585a6eb73baf7bb4185c1b4383a1195dca0f12e3946d737c7763bcef0b9544c515d905c5c44c5308b
+  languageName: node
+  linkType: hard
+
+"cheerio@npm:^1.0.0-rc.12":
+  version: 1.0.0-rc.12
+  resolution: "cheerio@npm:1.0.0-rc.12"
   dependencies:
-    css-select: ~1.2.0
-    dom-serializer: ~0.1.0
-    entities: ~1.1.1
-    htmlparser2: ^3.9.1
-    lodash.assignin: ^4.0.9
-    lodash.bind: ^4.1.4
-    lodash.defaults: ^4.0.1
-    lodash.filter: ^4.4.0
-    lodash.flatten: ^4.2.0
-    lodash.foreach: ^4.3.0
-    lodash.map: ^4.4.0
-    lodash.merge: ^4.4.0
-    lodash.pick: ^4.2.1
-    lodash.reduce: ^4.4.0
-    lodash.reject: ^4.4.0
-    lodash.some: ^4.4.0
-  checksum: b0a6cfa61eb7ae96e4cb8cfeeb14eb45bb790fa40098509268629c4cecca5b99124aabe6daa1154c497ac8def47bc3f9706cef5f0e8a6177a0c137d4bdaaf8b7
+    cheerio-select: ^2.1.0
+    dom-serializer: ^2.0.0
+    domhandler: ^5.0.3
+    domutils: ^3.0.1
+    htmlparser2: ^8.0.1
+    parse5: ^7.0.0
```

Comment:
I'm still very concerned about `parse5` not handling self-closing custom elements. Did you explicitly test that?

Reply:
This is one of the questions I tested: https://github.com/PrairieLearn/pl-cs233/blob/27fc8e3518876c7a8d9a6aac636d3280105901bc/questions/cmos_logic_2var/question.html. Does this cover what you have in mind?

Reply:
Here's a concrete example:

https://astexplorer.net/#/gist/9ca711354c80c0ba10bb6aefdc73f20d/062c008b0393b9ca3705f13e98e77c4b234f5597

```html
<div>
  <pl_figure />
  some more text
</div>
```

Notice how `some more text` is parsed as a child of `<pl_figure />`, which is obviously incorrect for our use case.

Reply:
Ah, ok, I tested with that scenario and it didn't show the text.

Reply:
I believe 204f651c8 resolves this issue (see https://cheerio.js.org/docs/advanced/configuring-cheerio#using-htmlparser2-for-html).

Reply:
Yeah, that's exactly what I had done (and reverted) in #1458.

Reply:
It's not exactly the same code. The code before #1458 invoked htmlparser2 directly, which is slightly different than what cheerio does in the slim output (htmlparser2 return XML, slim return HTML). I'm using cheerio itself to call htmlparser2.

Reply:
Ah, yes, I see that now - nice!